crates/router_derive/src/macros/generate_schema.rs

```diff
@@ -0,0 +1,199 @@
+            if let proc_macro2::TokenTree::Group(token_group) = token {
+                // Filter out punctuation marks
+                let res = token_group
+                    .stream()
+                    .into_iter()
+                    .filter_map(|token| {
+                        if let proc_macro2::TokenTree::Ident(ident) = token {
+                            Some(ident)
+                        } else {
+                            None
+                        }
+                    })
+                    .collect::<Vec<_>>();
+                Some(res)
+            } else {
+                None
+            }
+        })
+        .flatten()
+        .collect::<Vec<_>>()
+}
+
+/// Parse schemas from #[mandatory_in] attribute
+/// Example
+///
+/// #[mandatory_in(PaymentsCreateRequest, PaymentsUpdateRequest)]
+/// would return
+///
+/// [PaymentsCreateRequest, PaymentsUpdateRequest]
+fn get_schemas_to_create(attributes: Vec<syn::Attribute>) -> syn::Result<Vec<syn::Ident>> {
+    let attributes = attributes
+        .iter()
+        .filter(|attribute| {
+            attribute
+                .path
+                .segments
+                .first()
+                .map(|path_segment| path_segment.ident.to_string().eq("generate_schemas"))
+                .unwrap_or(false)
+        })
+        .flat_map(|attribute| get_ident_from_attr(attribute.tokens.to_owned()))
+        .collect::<Vec<_>>();
+
+    if attributes.is_empty() {
+        Err(syn::Error::new(
+            proc_macro2::Span::call_site(),
+            "At least one schema has to be passed in #[generate_schemas]",
+        ))?
+    } else {
+        Ok(attributes)
+    }
+}
+
+fn get_struct_fields(
+    data: syn::Data,
+) -> syn::Result<syn::punctuated::Punctuated<syn::Field, syn::token::Comma>> {
+    if let syn::Data::Struct(syn::DataStruct {
+        fields: syn::Fields::Named(syn::FieldsNamed { ref named, .. }),
+        ..
+    }) = data
+    {
+        Ok(named.to_owned())
+    } else {
+        Err(syn::Error::new(
+            proc_macro2::Span::call_site(),
+            "This macro cannot be used on structs with no fields",
+        ))
+    }
+}
+
+pub fn polymorphic_macro_derive_inner(
+    input: syn::DeriveInput,
+) -> syn::Result<proc_macro2::TokenStream> {
+    let schemas_to_create = get_schemas_to_create(input.attrs)
+        .map_err(|error| syn::Error::new(proc_macro2::Span::call_site(), error))?;
+
+    let fields = get_struct_fields(input.data)
+        .map_err(|error| syn::Error::new(proc_macro2::Span::call_site(), error))?;
+
+    // Go through all the fields and create a mapping of required fields for a schema
+    // PaymentsCreate -> ["amount","currency"]
+    // This will be stored in a hashset
+    // mandatory_hashset -> ((PaymentsCreate, amount), (PaymentsCreate,currency))
+
+    let mut mandatory_hashset = HashSet::<(syn::Ident, syn::Ident)>::new();
+    let mut other_fields_hm = HashMap::<syn::Field, Vec<syn::Attribute>>::new();
```

Comment:
Do you think it would help to use a `HashMap<NewTypeName, Vec<RequiredField>>` instead?

Reply:
The lookup for this will be `Is this field mandatory in this NewType`, this can be done in a single lookup using `Hashset<(NewType, field)>`, but if we convert this to hashmap, then first we would have to lookup the NewType and then in the vector we should find if the field is present, so Hashset will be better in this case.